test/data/copt_covdepsurf_example.yaml

```diff
@@ -0,0 +1,439 @@
+            heat-capacity-a: 0.020, heat-capacity-b: -0.156,
+            heat-capacity-unit: meV/K}
+    CO2_Pt: {model: Linear,
+             enthalpy: 0.48, enthalpy-unit: eV,
+             entropy: -0.031, entropy-unit: eV/K}
+    C_Pt: {model: Polynomial,
+           enthalpy-2nd-order: 3.86e4, enthalpy-unit: J/mol,
+           entropy-3rd-order: -1.26e4, entropy-unit: J/mol/K}
+    O_Pt: {model: Interpolative,
+           enthalpy-coverages: [0.0, 0.2, 0.5, 0.7, 0.9, 1.0],
+           enthalpies: [0.0, 0.5, 1.2, 2.7, 3.5, 4.0],
+           enthalpy-unit: kcal/mol,
+           entropy-coverages: [0.0, 0.5, 1.0],
+           entropies: [0.0, -0.7, -2.0],
+           entropy-unit: kcal/mol/K}
+- name: CO2_Pt
+  composition: {Pt: 1, C: 1, O: 2}
+  thermo:
+    model: NASA7
+    temperature-ranges: [100.0, 953.59, 5000.0]
+    data:
+    - [3.09268931, 5.2927875e-03, 8.99447254e-06, -1.67129326e-08, 7.04909721e-12,
+      -4.52128353e+04, -5.10056355]
+    - [5.99803259, 2.27485222e-03, -6.81092723e-07, 1.34432446e-10, -1.11509701e-14,
+      -4.61838262e+04, -21.1649844]
+  coverage-dependencies:
+    OC_Pt: {model: Piecewise-Linear,
+            enthalpy-low: 0.1, enthalpy-high: 0.3,
+            enthalpy-change: 0.5, enthalpy-unit: eV,
+            entropy-low: -0.05, entropy-high: -0.1,
+            entropy-change: 0.5, entropy-unit: eV/K}
+    CO2_Pt: {model: Linear,
+	     enthalpy: 0.73, enthalpy-unit: eV,
+	     entropy: -0.44, entropy-unit: eV/K}
+- name: C_Pt
+  composition: {Pt: 1, C: 1}
+  thermo:
+    model: NASA7
+    temperature-ranges: [100.0, 1401.13, 5000.0]
+    data:
+    - [-0.604856973, 0.0110908327, -1.34369594e-05, 7.18238215e-09, -1.40685044e-12,
+      6244.79345, 1.48480241]
+    - [3.08076078, -6.87465602e-04, 5.17589597e-07, -9.73022048e-11, 6.24243898e-15,
+      5335.31839, -17.0991414]
+  coverage-dependencies:
+    C_Pt: {model: Interpolative,
+           enthalpy-coverages: [0.0, 0.18, 0.67, 1.0],
+           enthalpies: [0.0, 0.12, 0.70, 0.82], 
+           enthalpy-unit: eV,
+           entropy-coverages: [0.0, 0.11, 0.33, 0.77, 1.0], 
+           entropies: [0.0, 0.05, -0.12, -0.40, -0.55],
+           entropy-unit: eV/K,
+           heat-capacity-a: 0.07, heat-capacity-b: -0.21,
+           heat-capacity-unit: meV/K}
+    O_Pt: {model: Linear,
+           enthalpy: 0.12, enthalpy-unit: eV,
+           entropy: 0.09, entropy-unit: eV/K}
+- name: O_Pt
+  composition: {Pt: 1, O: 1}
+  thermo:
+    model: NASA7
+    temperature-ranges: [100.0, 888.26, 5000.0]
+    data:
+    - [-0.759013067, 0.0189868498, -3.82473745e-05, 3.43558395e-08, -1.13974372e-11,
+      -1.72389494e+04, 1.76017396]
+    - [1.89893619, 2.03295425e-03, -1.19976574e-06, 2.32680659e-10, -1.53508282e-14,
+      -1.75144954e+04, -9.6410408]
+  coverage-dependencies:
+    O_Pt: {model: Polynomial,
+           enthalpy-1st-order: 0.14,
+           enthalpy-2nd-order: -1.74,
+           enthalpy-4th-order: 2.11, 
```

Comment:
I think it would make more sense to provide polynomial coefficients as a list, rather than with individually-named entries.